Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1367,29 +1369,40 @@ std::optional<std::vector<GCMemcardAnimationFrameRGBA8>> GCMemcard::ReadAnimRGBA
   return output;
 }
 
-bool GCMemcard::Format(u8* card_data, bool shift_jis, u16 SizeMb)
+bool GCMemcard::Format(u8* card_data, const CardFlashId& flash_id, u16 size_mbits, bool shift_jis,
+                       u32 rtc_bias, u32 sram_language, u64 format_time)
 {
   if (!card_data)
     return false;
-  memset(card_data, 0xFF, BLOCK_SIZE * 3);
-  memset(card_data + BLOCK_SIZE * 3, 0, BLOCK_SIZE * 2);
 
-  *((Header*)card_data) = Header(SLOT_A, SizeMb, shift_jis);
+  Header header(flash_id, size_mbits, shift_jis, rtc_bias, sram_language, format_time);
+  Directory dir;
+  BlockAlloc bat(size_mbits);
+
+  static_assert(std::is_trivially_copyable_v<Header>);
```

Comment:
These should be placed after the respective class/struct declarations in the header, so that the reader of the class knows that it's a requirement explicitly at the point of reading, rather than changes potentially being made and having it blow up on them in some other area of the code. Ditto for other occurrences.

Reply:
Yeah, I suppose that makes sense. Should I still leave them in the functions as well?

Reply:
Nah, just the one under the class declarations should be sufficient.